compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/fir.kt

```diff
@@ -857,6 +859,16 @@ internal fun FirAnnotation.resolvedScopeClassId(typeResolver: TypeResolveService
     ?: scopeArgument.resolvedClassArgumentTarget(typeResolver)?.classId
 }
 
+internal fun FirAnnotation.scopeName(session: FirSession): String? {
+  return resolvedScopeClassId()?.shortClassName?.identifier
+    ?: scopeArgument()
+      ?.argument
+      ?.toReference(session)
+      ?.expectAs<FirSimpleNamedReference>()
```

Comment:
nit: let's use `.expectAsOrNull()` since this may not be fully resolved yet in the IDE at the time of writing in the editor